crates/spin-test-sdk-macro/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+use heck::{ToKebabCase, ToSnakeCase, ToUpperCamelCase};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+
+const SPIN_TEST_NAME_PREFIX: &str = "spin_test_";
+
+#[proc_macro_attribute]
+pub fn spin_test(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let func_item = syn::parse_macro_input!(item as syn::ItemFn);
+    let func_name = &func_item.sig.ident;
+
+    let world_name = format!("{SPIN_TEST_NAME_PREFIX}{func_name}").to_kebab_case();
+    let world_text = generate_world(func_name);
```

Comment:
Nit: seems like the first thing we do in `generate_world` is recreate `world_name` so it might make more sense to just pass that down.